server/channels/app/oauth.go

```diff
@@ -1061,12 +1142,28 @@ func (a *App) GetAuthorizationServerMetadata(c request.CTX) (*model.Authorizatio
 }
 
 func (a *App) RegisterOAuthClient(c request.CTX, req *model.ClientRegistrationRequest, userID string) (*model.OAuthApp, *model.AppError) {
+	if !*a.Config().ServiceSettings.EnableOAuthServiceProvider {
+		return nil, model.NewAppError("RegisterOAuthClient", "api.oauth.register_oauth_app.turn_off.app_error", nil, "", http.StatusNotImplemented)
+	}
+
 	app := model.NewOAuthAppFromClientRegistration(req, userID)
 
-	rapp, err := a.CreateOAuthApp(app)
```

Comment:
Changed this because CreateOAuthApp was setup to always create a ClientSecret and unfortunately it's used in the plugin_api so I cannot change that behaviour